sql-plugin/src/main/scala/org/apache/spark/sql/rapids/collectionOperations.scala

```diff
@@ -407,13 +407,24 @@ case class GpuSortArray(base: Expression, ascendingOrder: Expression)
 
   override def doColumnar(lhs: GpuColumnVector, rhs: GpuScalar): cudf.ColumnVector = {
     val isDescending = isDescendingOrder(rhs)
-    lhs.getBase.listSortRows(isDescending, true)
+    val base = lhs.getBase()
+    dataType match {
+      case ArrayType(FloatType | DoubleType, _) =>
+        withResource(base.getChildColumnView(0)) {child =>
+          withResource(child.normalizeNANsAndZeros()) {normalizedChild =>
+            withResource(base.replaceListChild(normalizedChild)) {normalizedList =>
```

Comment:
@revans2 Does this code `base.replaceListChild` permanently modify the input column? If so, would this cause data corruption?

Reply:
Hi @ttnghia , i guess this function doesn't modify `base`, because `column vector` and `column view` are immutable.

In my opinion, `base.replaceListChild` creates a new columnVector to copy the `offsets buffer` of the `base` (this should be cheap because it just increases the reference count of the underlying bytes and copies the `offset` and `length` of the buffer). And it will update the child array to point to the `new_child`. (This should also be fast and cheap because it just increases the reference to the underlying array data).